specification/marketplacecatalog/resource-manager/Solutions/preview/2023-03-01-preview/solutions.json

```diff
@@ -0,0 +1,501 @@
+            "$ref": "./examples/GetUserSolutions.json"
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Marketplace/mysolutions/{solutionId}": {
+      "parameters": [
+        {
+          "$ref": "#/parameters/ApiVersionParameter"
+        },
+        {
+          "$ref": "#/parameters/SolutionIdParameter"
+        }
+      ],
+      "get": {
+        "tags": [
+          "UserSolutions"
+        ],
+        "operationId": "UserSolution_Get",
+        "description": "Get information about the user solution",
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/Solution"
+            }
+          },
+          "default": {
+            "description": "Microsoft.Marketplace error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "GetUserSolution": {
+            "$ref": "./examples/GetUserSolution.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "UserSolutions"
+        ],
+        "operationId": "UserSolution_CreateOrUpdate",
+        "description": "Creates or updates a user solutions",
+        "parameters": [
+          {
+            "in": "body",
+            "name": "payload",
+            "schema": {
+              "$ref": "#/definitions/CreateOrUpdateSolutionRequestPayload"
+            }
+          }
+        ],
+        "x-ms-examples": {
+          "CreateUserSolution": {
+            "$ref": "./examples/CreateUserSolution.json"
+          }
+        },
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/Solution"
+            }
+          },
+          "default": {
+            "description": "Microsoft.Marketplace error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "UserSolutions"
+        ],
+        "operationId": "RemoveUserSolution",
+        "description": "Removes a user solution",
+        "responses": {
+          "200": {
+            "description": "Success"
+          },
+          "204": {
+            "description": "No Content"
+          },
+          "default": {
+            "description": "Microsoft.Marketplace error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "RemoveUserSolution": {
+            "$ref": "./examples/RemoveUserSolution.json"
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Marketplace/mysolutions/{solutionId}/products/{productId}": {
```

Comment:
>products [](http://example.com/codeflow?start=63&length=8)

missing a GET on this URI to retrieve an individual product

Reply:
There is no need, products are publicly available through our public products APIs. The products collection is always required at its whole.

Reply:
removed this endpoint